src/core/models/querytriggers.js

```diff
@@ -0,0 +1,12 @@
+/** @module QueryTriggers */
+
+/**
+ * QueryTriggers is an ENUM of the possible triggers for a
+ * query update.
+ *
+ * @enum {string}
+ */
+export default {
+  INITIALIZE: 'initialize',
+  QUERY_PARAMETER: 'query-parameter'
```

Comment:
I think we should consider renaming `QUERY_PARAMETER`. In the context of our system, query is already a loaded term (search query), so I think it's a bit confusing for us to adopt another meaning for query here.